Getting it to work in VSCode?

I installed eslint-plugin-markdown in a project I'm editing with VScode and ran into issues

First I added the suggested overrides then I checked for errors in a markdown file, no errors show even thought I put intentional errors.

```
{
  ...
  "overrides": [
    {
      "files": [ "**/*.md" ],
      "parserOptions": {
        "ecmaFeatures": {
            "impliedStrict": true
        },
      },
      "rules": {
        "strict": "off"
      }
    }
  ]
}
```

I tried adding `"markdown"` to the `eslint.validate` setting as in in `<projectfolder>/.vscode/settings.json`

```
{
  "eslint.validate": [ "javascript", "html", "markdown" ]
}
```

But in that case VSCode seems be trying to lint the entire markdown file as JavaScript. I get errors on the entire file starting with the first line which is just normal text.

Any idea what I'm doing wrong?

I'm able to reproduce something like this with a configuration similar to yours, though I removed the `overrides`:

<details open>
<p><summary><code>.vscode/settings.json</code></summary></p>
<pre><code>{
    "eslint.validate": ["javascript", "markdown"]
}</code></pre>
</details>

<details open>
<p><summary><code>.eslintrc.js</code></summary></p>
<pre><code>module.exports = {
    plugins: [
    "markdown"
    ],
    extends: [
        "eslint:recommended"
    ],
    env: {
        browser: true
    }
}</code></pre>
</details>

In my case, lint rule failures are shown correctly:

![screenshot 2019-01-02 14 46 58](https://user-images.githubusercontent.com/1709537/50609291-42824a00-0e9d-11e9-9178-25cf1b796545.png)

But for some reason, syntax errors are all shown on the first line:

![screenshot 2019-01-02 14 49 20](https://user-images.githubusercontent.com/1709537/50609434-95f49800-0e9d-11e9-8da0-f9c40ca6e136.png)

It got the column (7) correct, but ignored the line. Running ESLint outside the integration returns the correct line and column:

```sh
$ node_modules/.bin/eslint README.md

/Users/brandon/code/eslint/markdown-test/README.md
  5:7  error  Parsing error: Unexpected token !

✖ 1 problem (1 error, 0 warnings)
```

I don't get any errors in a file that shouldn't have any:

![screenshot 2019-01-02 14 56 18](https://user-images.githubusercontent.com/1709537/50609740-9ccfda80-0e9e-11e9-9aab-2151768a1ece.png)

Does this match what you're seeing?

--- a/lib/processor.js
+++ b/lib/processor.js
@@ -279,7 +279,6 @@ function adjustBlock(block) {
     return function adjustMessage(message) {
 
         const lineInCode = message.line - leadingCommentLines;
-        const endLine = message.endLine - leadingCommentLines;
 
         if (lineInCode < 1) {
             return null;
@@ -287,9 +286,12 @@ function adjustBlock(block) {
 
         const out = {
             line: lineInCode + blockStart,
-            endLine: endLine ? endLine + blockStart : endLine,
             column: message.column + block.position.indent[lineInCode - 1] - 1
         };
+
+        if (Number.isInteger(message.endLine)) {
+            out.endLine = message.endLine - leadingCommentLines + blockStart;
+        }
 
         const adjustedFix = {};
 

--- a/tests/lib/plugin.js
+++ b/tests/lib/plugin.js
@@ -72,6 +72,20 @@ describe("plugin", () => {
         assert.strictEqual(report.results[0].messages[1].endLine, 8);
     });
 
+    it("doesn't add end locations to messages without them", () => {
+        const code = [
+            "```js",
+            "!@#$%^&*()",
+            "```"
+        ].join("\n");
+        const report = cli.executeOnText(code, "test.md");
+
+        assert.strictEqual(report.results.length, 1);
+        assert.strictEqual(report.results[0].messages.length, 1);
+        assert.notProperty(report.results[0].messages[0], "endLine");
+        assert.notProperty(report.results[0].messages[0], "endColumn");
+    });
+
     it("should emit correct line numbers with leading comments", () => {
         const code = [
             "# Hello, world!",

